### ISSUE TITLE
Nerd font is displayed incorrectly

### ISSUE THREAD

**Issue by wwWT000** [neovide/neovide#1944]
OS: Windows 11 10.0.22621
Neovide 0.10.4

The Nerd Font is displayed INCORRECTLY.

Steps to reproduce:
1.Open the Neovide
2.See error

**Comment by wwWT000** [comment]
In Neovide:
![屏幕截图 2023-07-25 124041](https://github.com/neovide/neovide/assets/101490485/bbb1dc16-1786-4559-a8ed-0252bb7385eb)
![屏幕截图 2023-07-25 124140](https://github.com/neovide/neovide/assets/101490485/030ac839-52a3-49af-bead-7e3c2d01eb33)
In the Terminal:
![屏幕截图 2023-07-25 124222](https://github.com/neovide/neovide/assets/101490485/3df7ee10-391c-412c-84cb-6eca587d84a7)
![屏幕截图 2023-07-25 124314](https://github.com/neovide/neovide/assets/101490485/b8543deb-ff85-41da-9f40-eb5c15522ffc)

**Comment by fredizzimo** [comment]
The logfile does not help much, since it's rotated, but you don't need to send it again. We just need know which nerd font you are using, and the version number. And your exact `guifont` string. And also your terminal configuration and the fallback fonts used there.

I suspect that it's one of the breaking changes in Nerd fonts 3.0 that is causing issues https://github.com/ryanoasis/nerd-fonts/releases/tag/v3.0.0, probaby the code point change, which has caused a lot of problems for Neovim users. And your terminal is probably picking those symbols up from a fallback font.

**Comment by wwWT000** [comment]
> 日志文件没有多大帮助，因为它已旋转，但您无需再次发送它。我们只需要知道您使用的是哪种书字体以及版本号。还有你的确切字符串。还有您的终端配置和那里使用的备用字体。`guifont`
> 
> 我怀疑这是书字体 3.0 中的重大更改之一，它导致了 https://github.com/ryanoasis/nerd-fonts/releases/tag/v3.0.0 的问题，导致代码点更改，这给 Neovim 用户带来了很多问题。您的终端可能正在从回退字体中获取这些符号。

Thank you VERY MUCH!
This may not be Neovide's problem

**Comment by wwWT000** [comment]
> The logfile does not help much, since it's rotated, but you don't need to send it again. We just need know which nerd font you are using, and the version number. And your exact `guifont` string. And also your terminal configuration and the fallback fonts used there.
> 
> I suspect that it's one of the breaking changes in Nerd fonts 3.0 that is causing issues https://github.com/ryanoasis/nerd-fonts/releases/tag/v3.0.0, probaby the code point change, which has caused a lot of problems for Neovim users. And your terminal is probably picking those symbols up from a fallback font.

Thank you VERY MUCH!
This may not be Neovide's problem

**Comment by WillEhrendreich** [comment]
same problem with Jetbrains Nerd Font Mono. is there any kind of workaround?

**Comment by fredizzimo** [comment]
It depends on which plugins you are using. Try to update those plugins, and search for issues in their repositories. Searches like "Neovim Nerd Fonts 3.0" also gives a lot of results.

It's possible that it's a Neovide issue, but those are the most likely issues.

**Comment by wwWT000** [comment]
> It depends on which plugins you are using. Try to update those plugins, and search for issues in their repositories. Searches like "Neovim Nerd Fonts 3.0" also gives a lot of results.
> 
> It's possible that it's a Neovide issue, but those are the most likely issues.

I "upgrade" my Nerd Font to 2.3.0, but the Bug still appear.
So weird i think

**Comment by fredizzimo** [comment]
I don’t think you installed it correctly. The missing icon in your screenshot is `nf-mdi-function`, which was renamed to `nf-md-function` in 3.0. You can check it by searching for`function` here https://www.nerdfonts.com/cheat-sheet, so it totally explains your problem.

Your guifont might be set wrong, making it use a different font or fallback than you think.

**Comment by wwWT000** [comment]
> I don’t think you installed it correctly. The missing icon in your screenshot is , which was renamed to in 3.0. You can check it by searching for here https://www.nerdfonts.com/cheat-sheet, so it totally explains your problem.`nf-mdi-function``nf-md-function``function`
> 
> Your guifont might be set wrong, making it use a different font or fallback than you think.



> same problem with Jetbrains Nerd Font Mono. is there any kind of workaround?



I "FIXED" this ISSUE now!! I installed the latest version of Nerd Font `JetBrains Mono Nerd Font 3.0.x`
then set it to Neovide's default font, `vim.o.guifont` all icons are displayed correctly now

But I don't think this comment should be closed.
Maybe someone else is having the same problem...


![image](https://github.com/neovide/neovide/assets/101490485/6cb0ea4b-fb17-415e-ab9c-eb06be79e1fe)

**Comment by fredizzimo** [comment]
@wwWT000. I think you might have to different FiraCode NF installed. First check in the windows settings font section, that it's listed only once, and has the correct version. Uninstall any versions that are wrong.

Then check the actual name with `set guifont=*` and make sure to select the right one.  Note that Nerdfonts 3.0 changed the naming of the fonts https://github.com/ryanoasis/nerd-fonts/releases/tag/v3.0.0

**Comment by fredizzimo** [comment]
I forgot to mention, that there's a bug in Windows 11, the font cache sometimes gets corrupted after a reboot. This affects not only Neovide, but also Windows terminal, and a lot of other programs. See this issue for more information https://github.com/microsoft/terminal/issues/3257

The workaround is to restart the ` Windows Font Cache Service`, there's also a new workaround in this comment https://github.com/microsoft/terminal/issues/3257#issuecomment-1619402456, but I have not tried that myself.

**Comment by wwWT000** [comment]
> I forgot to mention, that there's a bug in Windows 11, the font cache sometimes gets corrupted after a reboot. This affects not only Neovide, but also Windows terminal, and a lot of other programs. See this issue for more information [microsoft/terminal#3257](https://github.com/microsoft/terminal/issues/3257)
> 
> The workaround is to restart the , there's also a new workaround in this comment [microsoft/terminal#3257 (comment)](https://github.com/microsoft/terminal/issues/3257#issuecomment-1619402456), but I have not tried that myself.` Windows Font Cache Service`

THANK YOU VERY MUCH, @fredizzimo

**Comment by fredizzimo** [comment]
We should document this in the FAQ. Adding the "good first issue" tag if someone wants to do it.

**Comment by fredizzimo** [comment]
This is now officially tracked by Microsoft https://github.com/microsoft/terminal/issues/3257#issuecomment-1691425659
